### PR TITLE
Tickets/dm 46136

### PR DESCRIPTION
**Pull request by pothiers**

### CODE REVIEW

**Comment by pothiers** [comment]
Fixes for use with USDF.  Include filter of fields in recs to selected outfields.

**Review by Vebop — APPROVED** [review]
Good improvements

File: notebooks_tsqr/NightLog.ipynb [hunk 1]
```diff
@@ -72,8 +72,9 @@
     "usdf = 'https://usdf-rsp-dev.slac.stanford.edu'\n",
```
Comment (Vebop) [inline notebooks_tsqr/NightLog.ipynb:72]: <pre class="ql-syntax" spellcheck="false">Line #48.                    del rec[f]
</pre><p>Actually, I think you might get an error here when you delete from the list you are looping over? Unsure in this instance but I think it is dangerous to do.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_logging_and_reporting/pull/23/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (pothiers) [reply notebooks_tsqr/NightLog.ipynb:72]: <p>I agree its dangerous to delete from the list being looped over.  But that's not what is happening here.  The delete is deleting one element from a dictionary (not a rec). The dictionary is not being looped over.  In fact, your concern is why the nukefields is calculated, then iterated over.  It avoid the naive approach of iterating directly over rec.keys(), which would be the situation you warn against.</p>

**Pull request merged**